### PR TITLE
[mlflow/R] Allow client to infer context info when creating new run in fluent API.

### PR DESCRIPTION
**Pull request by tomasatdatabricks**
Context tags allow clients to infer extra tags from the environment.
These tags are automatically set when  creating new run in the fluent api.

### CODE REVIEW

**Comment by acroz** [comment]
Hey @tomasatdatabricks, I've been working with @aarondav to implement an extensible way to generate context-dependent tags on run creation in the Python client (see PR #930). Is there any way we can re-use the same logic to infer e.g. tags from the databricks notebook context without having to re-implement it in all of Python, Java and R, or is it going to be different enough that that's not possible?

**Comment by tomasatdatabricks** [comment]
Hey @acroz. I actually was not aware of your PR, thanks for pointing it out!

I think we need a separate logic in R to infer the context unfortunately. R can not just call into the python client because some of the context would be lost. It would be nice if the two worked in the same way though. 

I looked at you PR briefly and it looks like we differ fundamentally on where does the context come from. In this PR, the context is determined by the client (which is determined based on the tracking uri) whereas in your case you allow number of contexts independent on the tracking uri which I think makes sense. 

I think for now I would prefer to merge this one as is and wait until your PR is merged to make changes. (I don't think the two clients need to get updated at the same time.)

File: mlflow/R/mlflow/R/databricks-utils.R
Comment (smurching) [inline mlflow/R/mlflow/R/databricks-utils.R]: One observation entirely orthogonal to this PR, I guess we don't set `source_type` elsewhere in the R client - we might want to consider defaulting `source_type` to `LOCAL` [as the Python client does](https://github.com/mlflow/mlflow/blob/master/mlflow/tracking/fluent.py#L154), since the runs UI renders an icon conditional on source type (i.e. add a `get_source_type` function in `tracking-utils.R` and then invoke it in `mlflow_get_run_context.default` within `tracking-client.R`. Will make a GitHub issue to track it separately

**Review by smurching — APPROVED** [review]
LGTM

File: mlflow/R/mlflow/R/databricks-utils.R
Comment (tomasatdatabricks) [reply mlflow/R/mlflow/R/databricks-utils.R]: That's a good point!

**Comment by acroz** [comment]
> I think we need a separate logic in R to infer the context unfortunately. R can not just call into the python client because some of the context would be lost. It would be nice if the two worked in the same way though.

Yeah, seems fair.

> I think for now I would prefer to merge this one as is and wait until your PR is merged to make changes. (I don't think the two clients need to get updated at the same time.)

This sounds good. There are further changes beyond what's in #926 and #930 I'd like to make to tidy up the context inference logic (e.g. some tags are set in the fluent layer, some in the client layer, and one (the run name) in the tracking store implementations themselves) - perhaps we should finish those changes and then look at reconciling the Python and R logic?

**Review by kevinykuo — APPROVED** [review]
LGTM!

**Review by javierluraschi — APPROVED** [review]

**Comment by tomasatdatabricks** [comment]
@acroz Sounds good!

**Pull request merged**
